ops/testing.py

```diff
@@ -1507,32 +1605,32 @@ def _storage_remove(self, storage_id: str):
         index = int(index)
         self._storage_list[name].pop(index, None)
 
-    def action_get(self):
-        raise NotImplementedError(self.action_get)
+    def action_get(self):  # type:ignore
+        raise NotImplementedError(self.action_get)  # type:ignore
 
-    def action_set(self, results):
-        raise NotImplementedError(self.action_set)
+    def action_set(self, results):  # type:ignore
+        raise NotImplementedError(self.action_set)  # type:ignore
 
-    def action_log(self, message):
-        raise NotImplementedError(self.action_log)
+    def action_log(self, message):  # type:ignore
+        raise NotImplementedError(self.action_log)  # type:ignore
 
-    def action_fail(self, message=''):
-        raise NotImplementedError(self.action_fail)
+    def action_fail(self, message=''):  # type:ignore
+        raise NotImplementedError(self.action_fail)  # type:ignore
 
-    def network_get(self, endpoint_name, relation_id=None):
-        raise NotImplementedError(self.network_get)
+    def network_get(self, endpoint_name, relation_id=None):  # type:ignore
+        raise NotImplementedError(self.network_get)  # type:ignore
 
-    def add_metrics(self, metrics, labels=None):
-        raise NotImplementedError(self.add_metrics)
+    def add_metrics(self, metrics, labels=None):  # type:ignore
+        raise NotImplementedError(self.add_metrics)  # type:ignore
 
     @classmethod
-    def log_split(cls, message, max_len):
-        raise NotImplementedError(cls.log_split)
+    def log_split(cls, message, max_len):  # type:ignore
+        raise NotImplementedError(cls.log_split)  # type:ignore
 
-    def juju_log(self, level, msg):
-        raise NotImplementedError(self.juju_log)
+    def juju_log(self, level, msg):  # type:ignore
+        raise NotImplementedError(self.juju_log)  # type:ignore
 
-    def get_pebble(self, socket_path: str) -> 'pebble.Client':
+    def get_pebble(self, socket_path: str) -> '_TestingPebbleClient':
```

Comment:
Same here with return type annotations.  Also the max_len looks like it should have a default of MAX_LOG_LING_LEN to be consistent with model.py impl.